examples/client_async_calls.py

```diff
@@ -169,7 +169,7 @@ async def async_write_registers_mypy(client):
     await client.write_registers(1, regs2, slave=SLAVE)
     rr = await client.read_holding_registers(1, count=len(regs2), slave=SLAVE)
     assert not rr.isError()  # test that call was OK
-    assert rr.registers == regs2
+    assert rr.registers == [int.from_bytes(i, byteorder="big") for i in regs2]
```

Comment:
This seems wrong, registers in responses are always int never bytes.

Reply:
Because the test was broken when the values were copied "appropriately". After the call, `regs2` is no longer a list of bytes, it's a list of ints. The reason the comparison works is because of that mutation. if regs2 is not mutated by the call, it continues to be a list of bytes, which you cannot compare to a list of ints.

Your latest commit now changes how all of this works by trying to inform the caller that their data will potentially be corrupted by this call. If they do not want that, they will have to send a copy of their data every time they make an API call which sounds expensive